packages/experimental/Badge/src/PresenceBadge/PresenceBadge.tsx

```diff
@@ -0,0 +1,67 @@
+/** @jsx withSlots */
+import { Badge } from '../Badge';
+import { presenceBadgeName, PresenceBadgeType, PresenceBadgeProps, Presence } from './PresenceBadge.types';
+import { compose, withSlots, mergeProps, stagedComponent } from '@fluentui-react-native/framework';
+import { presenceIcons } from './presenceIcons';
+import { SvgXml } from 'react-native-svg';
+import { useBadge } from '../useBadge';
+
+function getIcon(presence: Presence, oof: boolean) {
+  switch (presence) {
+    case 'available':
+    default:
+      return oof ? presenceIcons.availableOof : presenceIcons.available;
+    case 'away':
+      return oof ? presenceIcons.oof : presenceIcons.away;
+    case 'busy':
+      return oof ? presenceIcons.busyOof : presenceIcons.busy;
+    case 'DND':
+      return oof ? presenceIcons.dndOof : presenceIcons.dnd;
+    case 'offline':
+      return presenceIcons.offline;
+    case 'OOF':
+      return presenceIcons.oof;
+  }
+}
+
+export const PresenceBadge = compose<PresenceBadgeType>({
+  displayName: presenceBadgeName,
+  slots: {
+    badge: Badge,
+  },
+  render: (userProps: PresenceBadgeProps) => {
+    const badge = useBadge(userProps);
+    const iconXml = getIcon(userProps.presence, userProps.oof);
+    const CustomBadge = Badge.customize({
+      borderWidth: 0,
+      padding: 0,
+    });
+
+    return (final: PresenceBadgeProps) => {
+      const { appearance = 'outline', ...mergedProps } = mergeProps(badge, final);
+      return (
+        <CustomBadge appearance={appearance} {...mergedProps}>
+          <SvgXml xml={iconXml} />
+        </CustomBadge>
+      );
+    };
+  },
+});
+
+export const PresenceBadgeStaged = stagedComponent((props: PresenceBadgeProps) => {
```

Comment:
I think the question with using stagedComponent is that stagedComponent does not allow for customizable/compose to be called to extend the component. Whether or not this is necessary is something we should probably ask PM.

Can you explain what you mean by compose won't let you style the inner badge using tokens? can you give a code example so I can understand the issue better?

Reply:
It's the same issue we had with MenuButton if we wanted to customize Button component which is part of MenuButton

Reply:
So now that I understand the framework a bit more, I /think/ you can accomplish customizing the inner Badge by running customize on Badge and then passing the customized Badge as the new slot for the badge slot of PresenceBadge using compose. So it'd be something like:

```ts
const CustomBadge = Badge.customize({ /*modified tokens for inner Badge*/ });
const ComposedPresenceBadge = PresenceBadge.compose({ slots: { badge: CustomBadge },  slotProps: { /*new slotProps*/ } });
const CustomPresenceBadge = ComposedPresenceBadge.customize({ /* tokens for PresenceBadge */} )
```

It's not very user friendly though, maybe compressible the is the way to go instead.

Reply:
I think for now we should at least remove the stagedComponent version, and maybe get this checked in.

Reply:
I removed it